syntax = "proto3";

message JoinRequest {
    Address address = 1;
    Specs specs = 2;
}

message Address {
    string IP = 1;
    string port = 2;
}

message Specs {
    string CPU = 1;
    string RAM = 2;
    string storage = 3;
}

message JoinResponse {
    repeated Address existing_peers = 1;
}

service BootstrapService {
    rpc JoinNetwork(JoinRequest) returns (JoinResponse);
}
